### ISSUE TITLE
extension-stmt conformance

### ISSUE THREAD

**Issue by abierman** [netmod-wg/yang-next#71]
The conformance requirements and capabilities discovery for extension statements is broken
and implementations do not follow it.

If a module contains extensions and a server advertises the implemented module
in the YANG library, then the server must implement all extensions in the module.

There is no way to make if-feature a required sub-statement of an extension.
There is no way to specify a deviation to an extension, even to deviate not-supported.

Not all extensions are meant for the server, and the server implementor
just chooses some extensions in an ad-hoc, un-advertised manner, so the client tools
do not know for sure which extensions will be supported or not

**Comment by kwatsen** [comment]
From Wed meeting: don't fully understand the problem.  Is this a duplicate of the "critical extension" issue?   #49

**Comment by abierman** [comment]
No

There is no way to make an extension optional.
There is no way to specify that the server does not implement an extension.
The ticket is very clear on the issues

**Comment by abierman** [comment]
Consider NACM.
If a server claims conformance to NACM (listed as an implemented module in yanglib)
then is it allowed to ignore the default-deny-all extension? (server MAY ignore any extension)

If no, how does a server indicate that this extension is not supported?
If yes, then how can an operator rely on NACM extensions?

**Comment by abierman** [comment]
this issue should be closed.
the YANG Packages work should address any conformance/capabilities for some use-case

**Comment by QiufangMa** [comment]
Does the following deviation work?
     deviation /a/b {
       deviate delete {
         myext:XXX "YYY";
       }
     }

**Comment by janlindblad** [comment]
I agree that YANG Packages would be a good place to deal with conformance in general. For most part, I think the use/conformance requirements for extension statements are handled ok in RFC text (e.g. NACM implementors have to support it because the RFC speaks about them), but maybe there should be a way to find out in YANG Library or Packages etc.

**Comment by kwatsen** [comment]
Kent: config-false lists, YL, and/or packages could all be used to indicate which extensions are supported.
Andy: deviate-delete seems like a easy fix
Kent: are we trying to define conformance for the extension or uses of the extension

**Comment by kwatsen** [comment]
Hi Andy, can you doublecheck this issue?

I just searched RFC 7950 and nowhere do I see that the module defining the extension has to be implemented.  All the text I found just says that the extension statements can be imported and used by other modules.  Never do I see anything regarding the implementation-status of the module.

**Comment by abierman** [comment]
Extensions are external statements.
RFC 7950 does not place any implementation requirements on external statements
except a tool must be able to skip the external statement.

The module defining the extension needs to define any implementation requirements.

This is not related to RFC 7950 conformance.
Even in YANG 2.0 any tool MAY skip any external statement.

If a tool wants to claim conformance to a module that defines an extension,
then it is expected to follow the implementation requirements in that module
